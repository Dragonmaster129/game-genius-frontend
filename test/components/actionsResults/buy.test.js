import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import Buy from "../../../src/components/actionsResults/buy";
import sampledata from "../../sampledata";
import endData from "../../endDataBuy";

let cash = 3950;
const setcash = (value) => {
  console.log("in setcash");
  cash = value;
};
let data = sampledata;
const setdata = (value) => {
  console.log("in setdata");
  data = value;
};
const onChange = (value) => {
  console.log("in onChange");
};

test("renders", () => {
  render(
    <Buy
      data={data}
      setdata={setdata}
      submitted={onChange}
      cash={cash}
      setcash={setcash}
    />
  );
  let buy = screen.getByRole("button");
  expect(buy).toBeDefined();
});

test("buy an item", async () => {
  render(
    <Buy
      data={data}
      setdata={setdata}
      submitted={onChange}
      cash={cash}
      setcash={setcash}
    />
  );
  let header = screen.getByTestId("REALESTATE");
  fireEvent.click(header);
  let cost = screen.getByTestId("cost");
  let mortgage = screen.getByTestId("mortgage");
  let cashFlow = screen.getByTestId("cashFlow");
  let STARTERHOUSE = screen.getByTestId("STARTERHOUSE");
  let downpay = screen.getByTestId("downpay");
  fireEvent.click(STARTERHOUSE);
  fireEvent.change(cost, { target: { value: 52000 } });
  fireEvent.change(mortgage, { target: { value: 50000 } });
  fireEvent.change(downpay, { target: { value: 2000 } });
  fireEvent.change(cashFlow, { target: { value: 20 } });
  let BUY = screen.getByRole("button");
  fireEvent.click(BUY);
  expect(data).toEqual(endData);
});
