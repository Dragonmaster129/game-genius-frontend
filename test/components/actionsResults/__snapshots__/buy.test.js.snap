// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buy component is the same 1`] = `
<div
  className="buy"
>
  <select
    className="buy-choice"
    name="select-item-to-buy"
  >
    <option
      onClick={[Function]}
      value="NONE"
    >
      NONE
    </option>
    <option
      onClick={[Function]}
      value="STOCK"
    >
      STOCK
    </option>
    <option
      onClick={[Function]}
      value="REALESTATE"
    >
      REALESTATE
    </option>
    <option
      onClick={[Function]}
      value="D2Y"
    >
      D2Y
    </option>
    <option
      onClick={[Function]}
      value="LAND"
    >
      LAND
    </option>
    <option
      onClick={[Function]}
      value="BUSINESS"
    >
      BUSINESS
    </option>
  </select>
  <div>
    <h1>
      NONE
    </h1>
    <form
      className="v"
    >
      <label>
        Spending:
      </label>
      0
      <label>
        <h3>
          Cash:
        </h3>
      </label>
      <h3>
        3,950
      </h3>
      
      <button
        onClick={[Function]}
      >
        BUY
      </button>
    </form>
  </div>
</div>
`;
